handlers/dashboard.go

```diff
@@ -235,13 +235,19 @@ func getNextWithdrawalRow(queryValidators []uint64) ([][]interface{}, error) {
 			continue
 		}
 		for _, v := range validatorsDb {
-			if v.Index == balanceIndex {
-				if (balance[0].Balance > 0 && v.WithdrawableEpoch <= epoch) ||
-					(balance[0].EffectiveBalance == utils.Config.Chain.Config.MaxEffectiveBalance && balance[0].Balance > utils.Config.Chain.Config.MaxEffectiveBalance) {
-					// this validator is withdrawable, check if it is the next one
-					if nextValidator == nil || v.Index > *stats.LatestValidatorWithdrawalIndex {
-						nextValidator = v
-						nextValidator.Balance = balance[0].Balance
+			if v.Index != balanceIndex {
+				continue
+			}
+
+			if (balance[0].Balance > 0 && v.WithdrawableEpoch <= epoch) ||
+				(balance[0].EffectiveBalance == utils.Config.Chain.Config.MaxEffectiveBalance && balance[0].Balance > utils.Config.Chain.Config.MaxEffectiveBalance) {
+				// this validator is eligible for withdrawal, check if it is the next one
+				if nextValidator == nil || v.Index > *stats.LatestValidatorWithdrawalIndex {
+					nextValidator = v
+					nextValidator.Balance = balance[0].Balance
+					if nextValidator.Index > *stats.LatestValidatorWithdrawalIndex {
+						// the first validator after the cursor has to be the next validator
+						break
```

Comment:
As discussed this break only leaves the inner for loop and not both so the `nextValidator` still gets overwritten.

Reply:
Good find. Fixed.